python/pyspark/mlv2/classification.py

```diff
@@ -0,0 +1,309 @@
+    import torch.optim as optim
+
+    # TODO: add a setting seed param.
+    torch.manual_seed(seed)
+
+    # TODO: support training on GPU
+    # TODO: support L1 / L2 regularization
+    torch.distributed.init_process_group("gloo")
+
+    ddp_model = DDP(
+        _LinearNet(num_features=num_features, num_classes=num_classes, bias=fit_intercept)
+    )
+
+    loss_fn = torch_nn.CrossEntropyLoss()
+
+    optimizer = optim.SGD(ddp_model.parameters(), lr=learning_rate, momentum=momentum)
+    data_loader = _get_spark_partition_data_loader(
+        num_samples_per_worker,
+        batch_size,
+        num_workers=0,
+        prefetch_factor=None,
+    )
+    for i in range(max_iter):
+        ddp_model.train()
+
+        step_count = 0
+
+        loss_sum = 0.0
+        for x, target in data_loader:
+            optimizer.zero_grad()
+            output = ddp_model(x.to(torch.float32))
+            loss = loss_fn(output, target.to(torch.long))
+            loss.backward()
+            loss_sum += loss.detach().numpy()
+            optimizer.step()
+            step_count += 1
+
+        # TODO: early stopping
+        #  When each epoch ends, computes loss on validation dataset and compare
+        #  current epoch validation loss with last epoch validation loss, if
+        #  less than provided `tol`, stop training.
+
+        if torch.distributed.get_rank() == 0:
+            print(f"Progress: train epoch {i + 1} completes, train loss = {loss_sum / step_count}")
+
+    if torch.distributed.get_rank() == 0:
+        return ddp_model.module.state_dict()
+
+    return None
+
+
+@inherit_doc
+class LogisticRegression(Predictor["LogisticRegressionModel"], _LogisticRegressionParams):
+    """
+    Logistic regression estimator.
+
+    .. versionadded:: 3.5.0
+    """
+
+    def __init__(
+        self,
+        *,
+        featuresCol: str = "features",
+        labelCol: str = "label",
+        predictionCol: str = "prediction",
+        probabilityCol: str = "probability",
+        maxIter: int = 100,
+        tol: float = 1e-6,
+        numTrainWorkers: int = 1,
+        batchSize: int = 32,
+        learningRate: float = 0.001,
+        momentum: float = 0.9,
+        seed: int = 0,
+    ):
+        super(_LogisticRegressionParams, self).__init__()
+        self._set(
+            featuresCol=featuresCol,
+            labelCol=labelCol,
+            predictionCol=predictionCol,
+            probabilityCol=probabilityCol,
+            maxIter=maxIter,
+            tol=tol,
+            numTrainWorkers=numTrainWorkers,
+            batchSize=batchSize,
+            learningRate=learningRate,
+            momentum=momentum,
+            seed=seed,
+        )
+
+    def _fit(self, dataset: Union[DataFrame, pd.DataFrame]) -> "LogisticRegressionModel":
+        if isinstance(dataset, pd.DataFrame):
+            # TODO: support pandas dataframe fitting
+            raise NotImplementedError("Fitting pandas dataframe is not supported yet.")
```

Comment:
should use `PySparkNotImplementedError` instead

Reply:
I think `PySparkXXXError` is only used in pyspark core code.